python/cuda_cccl/tests/compute/examples/struct/nested_struct_tuple_construction.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# example-begin
+"""
+Example showing tuple syntax for constructing nested gpu_struct types.
+
+When working with nested structs in device functions, you can use tuple syntax
+as a convenient shorthand for constructing the nested struct values. This can
+make code more concise while maintaining the same functionality.
+"""
+
+import cupy as cp
+import numpy as np
+
+import cuda.compute
+from cuda.compute import gpu_struct
+
+
+# Define nested structs
+@gpu_struct
+class Stats:
+    count: np.int32
+    sum: np.float32
+
+
+@gpu_struct
+class DataPoint:
+    value: np.int64
+    stats: Stats
+
+
+def sum_with_tuples(d1, d2):
+    """
+    Reduction operation using tuple syntax for nested struct construction.
+
+    Instead of writing: Stats(d1.stats.count + d2.stats.count, ...)
+    We can use tuple syntax: (d1.stats.count + d2.stats.count, ...)
+    """
+    return DataPoint(
+        d1.value + d2.value,
+        # Tuple syntax for constructing the nested Stats struct
+        (d1.stats.count + d2.stats.count, d1.stats.sum + d2.stats.sum),
+    )
```

Comment:
Question: is it possible to also get rid of the `DataPoint` constructor call? It might be helpful to keep it for readability, but I am just curious.